core/src/androidTest/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/YouTubePlayerViewClickTest.kt

```diff
@@ -0,0 +1,128 @@
+package com.pierfrancescosoffritti.androidyoutubeplayer.core
+
+import android.app.Activity
+import android.view.MotionEvent
+import android.view.View
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.*
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.filters.LargeTest
+import androidx.test.rule.ActivityTestRule
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.views.YouTubePlayerView
+import com.pierfrancescosoffritti.androidyoutubeplayer.core.testActivity.TestActivity
+import com.pierfrancescosoffritti.androidyoutubeplayer.test.R
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.verify
+import java.util.*
+
+@LargeTest
+class YouTubePlayerViewClickTest {
+
+    private lateinit var testActivity: Activity
+    private lateinit var youTubePlayerView: YouTubePlayerView
+
+    @Rule
+    @JvmField
+    val activityRule = ActivityTestRule(TestActivity::class.java)
+
+    @Before
+    fun setup() {
+        testActivity = activityRule.activity
+        youTubePlayerView = testActivity.findViewById(R.id.youtube_player_view)
+    }
+
+    @Test
+    fun testSingleClickListener() {
+        // Prepare
+        val mockClickListener = mock(View.OnClickListener::class.java)
+        youTubePlayerView.setOnClickListener(mockClickListener)
+
+        // Act
+        onView(withId(R.id.youtube_player_view)).perform(click())
+
+        // Assert
+        verify(mockClickListener).onClick(youTubePlayerView)
+    }
+
+    @Test
+    fun testLongClickListener() {
+        // Prepare
+        val mockLongClickListener = mock(View.OnLongClickListener::class.java)
+        youTubePlayerView.setOnLongClickListener(mockLongClickListener)
+
+        // Act
+        onView(withId(R.id.youtube_player_view)).perform(longClick())
+
+        // Assert
+        verify(mockLongClickListener).onLongClick(youTubePlayerView)
+    }
+
+    @Test
+    fun testTouchSwipeDownListener() {
+        // Prepare
+        val touchEventQueue = LinkedList<MotionEvent>()
+        var onTouchWorks = false
+        youTubePlayerView.setOnTouchListener { _, motionEvent ->
+            touchEventQueue.add(MotionEvent.obtain(motionEvent))
+            onTouchWorks = true
+            false
+        }
+
+        // Act
+        onView(withId(R.id.youtube_player_view)).perform(swipeDown())
+
+        // Assert
+        assertTrue(onTouchWorks)
+
+        lateinit var prevMotionEvent: MotionEvent
```

Comment:
I'm confused, don't you get a compilation error using `lateinit` here?

Reply:
nope, it works

Reply:
Interesting. Let's make it nullable instead:
`var prevMotionEvent: MotionEvent? = null`

Reply:
sure thing!